sql_file3.sql

```diff
@@ -0,0 +1,705 @@
+    WHEN '00000AK47S' THEN 'Product details for 00000AK47S'
+    WHEN '00000B2ORH' THEN 'Product details for 00000B2ORH'
+    ELSE ProductDetails  
+END
+WHERE ProductID IN ('000007JDFT', '00000A1YZT', '00000AK47S', '00000B2ORH');
+
+
+UPDATE Products
+SET ProductDetails = LOWER(ProductDetails);
+
+
+-- 29 Remove trailing spaces from the string 'SQL Server   '.
+
+SELECT RTRIM('SQL Server    ') AS TrimmedString
+
+-- 30  Write a query to remove trailing spaces from all email addresses in an 'Employees' table
+
+select * from Employees;
+
+ALTER TABLE Employees
+ADD Email VARCHAR(50);
+
+UPDATE Employees
+SET Email = CASE EmployeeID
+    WHEN 1 THEN 'leodas@gmail.com   '
+    WHEN 2 THEN 'mukhil.mp@gmail.com      '
+    WHEN 3 THEN 'allenjose@gmail.com   '
+    ELSE Email  
+END
+WHERE EmployeeID IN (1,2,3);
+
+UPDATE Employees
+SET Email = RTRIM(Email);
+
+-- 31 Remove leading spaces from the string '   SQL Server'
+
+
+ SELECT LTRIM('   SQL Server') AS TrimmedString;
+
+ -- 32  From a 'Comments' table, write a query to remove leading spaces from all comment texts.
+ CREATE TABLE COMMENTS(
+ Comments VARCHAR(300));
+
+ INSERT INTO COMMENTS (Comments) VALUES
+('        This is a great product!'),
+('  I had an issue with the installation process.'),
+('Customer service was very helpful.'),
+('   Delivery was delayed by two days, but overall satisfied.'),
+('        Highly recommend to anyone looking for quality.'),
+('The user manual could be more detailed.'),
+('  Packaging was excellent, and the product arrived in good condition.'),
+('   The product exceeded my expectations. Will buy again.'),
+(' Had a minor issue, but it was resolved quickly.'),
+('     Great value for the price.');
+
+SELECT * FROM Comments;
+
+UPDATE Comments
+SET Comments= LTRIM(Comments);
+
+
+-- 33 Display the current date in the format 'dd-MM-yyyy'
+
+SELECT FORMAT(GETDATE(),'dd-MM-yyyy') AS CurrentDate;
+
+-- 34 From an 'Orders' table with an 'OrderTotal' column, display the total as a currency with 2 decimal places
+
+CREATE TABLE Orders2 (
+    OrderID INT PRIMARY KEY,
+    OrderTotal DECIMAL(10, 2)  -- 10 digits total, 2 decimal places
+);
+
+INSERT INTO Orders2 (OrderID, OrderTotal) VALUES
+(1, 1234.56),
+(2, 789.99),
+(3, 1500.00),
+(4, 99.95),
+(5, 2499.49);
+
+SELECT FORMAT(OrderTotal, 'C', 'en-US') AS FormattedTotal
+FROM Orders2;
+
+
+--35 Separate the string 'apple,banana,cherry' into individual fruits
+
+SELECT value AS Fruit
+FROM STRING_SPLIT('apple,banana,cherry', ',');
+
+-- 36 From a 'Skills' table with a 'SkillList' column containing comma-separated skills, write a query to create a row for each individual skill.
+CREATE TABLE Skills (
+    SkillID INT PRIMARY KEY,
+    SkillList VARCHAR(255)
+);
+
+INSERT INTO Skills (SkillID, SkillList) VALUES
+(1, 'SQL,Python,Data Analysis'),
+(2, 'Java,C++,JavaScript'),
+(3, 'HTML,CSS,React');
+
+
+SELECT SkillID, VALUE AS Skill
+FROM Skills
+CROSS APPLY STRING_SPLIT(SkillList, ',');
```

Comment:
Difference cross join and cross apply ?

Reply:
CROSS JOIN creates all possible combinations of rows from the two tables, regardless of any conditions.
 CROSS APPLY  is used when you need to apply a table-valued function or derived table to each row of the outer table